apps/calendar/js/views/modify_event.js

```diff
@@ -357,114 +256,69 @@ ModifyEvent.prototype = {
-      // it via css during that time period
-      list.add(self.PROGRESS);
-
-      var moveDate = self.event.startDate;
-
-      provider[method](self.event.data, function(err) {
-        list.remove(self.PROGRESS);
-
-        if (err) {
-          self.showErrors(err);
-          return;
-        }
-
-        // move the position in the calendar to the added/edited day
-        core.timeController.move(moveDate);
-        // order is important the above method triggers the building
-        // of the dom elements so selectedDay must come after.
-        core.timeController.selectedDay = moveDate;
-
-        // we pass the date so we are able to scroll to the event on the
-        // day/week views
-        var state = {
-          eventStartHour: moveDate.getHours()
-        };
-
-        if (method === 'updateEvent') {
-          // If we edit a view our history stack looks like:
-          //   /week -> /event/view -> /event/save -> /event/view
-          // We need to return all the way to the top of the stack
-          // We can remove this once we have a history stack
-          viewFactory.get('ViewEvent', function(view) {
-            router.go(view.returnTop(), state);
-          });
-
-          return;
-        }
-
-        router.go(self.returnTo(), state);
-      });
+    } catch(err) {
+      this.showErrors(err);
     }
-  },
+
+    this.element.classList.remove(this.PROGRESS);
+  }),
 
   /**
    * Deletes current record if provider is present and has the capability.
    */
-  deleteRecord: function(event) {
+  deleteRecord: co.wrap(function *(event) {
     if (event) {
       event.preventDefault();
     }
 
-    if (this.isSaved()) {
-      var self = this;
-      var handleDelete = function me_handleDelete() {
-        self.provider.deleteEvent(self.event.data, function(err) {
-          if (err) {
-            self.showErrors(err);
-            return;
-          }
-
-          // If we edit a view our history stack looks like:
-          //   /week -> /event/view -> /event/save -> /event/view
-          // We need to return all the way to the top of the stack
-          // We can remove this once we have a history stack
-          viewFactory.get('ViewEvent', function(view) {
-            router.go(view.returnTop());
-          });
-        });
-      };
-
-      this.provider.eventCapabilities(this.event.data, function(err, caps) {
-        if (err) {
-          return console.error('Error fetching event capabilities', this.event);
-        }
+    if (!this.isSaved()) {
+      return;
+    }
 
-        if (caps.canDelete) {
-          handleDelete();
-        }
+    try {
+      yield core.bridge.deleteEvent(this.event);
+      // If we edit a view our history stack looks like:
+      //   /week -> /event/view -> /event/save -> /event/view
+      // We need to return all the way to the top of the stack
+      // We can remove this once we have a history stack
+      viewFactory.get('ViewEvent', view => {
```

Comment:
I wonder if we could have a small module dedicated to fetching and rendering / doing something with a view to clean up the calling code a bit